skbio/alignment/tests/test_tabular_msa.py

```diff
@@ -1015,5 +1088,101 @@ def test_from_dict_to_dict_roundtrip(self):
         self.assertIs(d1[42], d2[42])
 
 
+class TestAppend(unittest.TestCase):
+    def setUp(self):
+        self.msa = TabularMSA([DNA('ACGT'), DNA('TGCA')])
+        self.append_seq = DNA('GGGG')
+
+        self.msa_with_keys = TabularMSA([DNA(''), DNA('')], keys=['a', 'b'])
+        self.append_seq_with_keys = DNA('')
+        self.append_key = 'c'
+        self.msa_with_keys_after_append = \
+            TabularMSA([DNA(''), DNA(''), self.append_seq_with_keys],
+                       keys=['a', 'b', self.append_key])
+
+    def test_to_empty_msa(self):
+        expected = TabularMSA([DNA('ACGT')])
+        msa = TabularMSA([])
+        msa.append(DNA('ACGT'))
+        self.assertEqual(msa, expected)
+
+    def test_to_empty_msa_invalid_dtype(self):
+        msa = TabularMSA([])
+        with six.assertRaisesRegex(self, TypeError,
+                                   'sequence.*alphabet.*Sequence'):
+            msa.append(Sequence(''))
```

Comment:
In these error-case tests can you check that the original MSA state isn't mangled?